### PR TITLE
[fix] resource_monitor credit quota as float64, not int

### PR DESCRIPTION
**Pull request by chauvm**
Issue: https://github.com/chanzuckerberg/terraform-provider-snowflake/issues/109

Note that if a float has more than 6 trailing numbers after the decimal point, `%f` rounds it up to have exactly 6. If fewer than 6 trailing numbers are provided, it appends zeros, for example, `66.00` becomes `66.000000`, which I think is fine?

### CODE REVIEW

File: pkg/resources/resource_monitor.go [hunk 1]
```diff
@@ -183,9 +183,9 @@ func ReadResourceMonitor(data *schema.ResourceData, meta interface{}) error {
 
 	// Credit quota is the only integer
```
Comment (edulop91) [inline pkg/resources/resource_monitor.go]: not an integer, maybe worth fixing that comment

File: pkg/snowflake/resource_monitor_test.go [hunk 2]
```diff
@@ -22,14 +22,14 @@ func TestResourceMonitor(t *testing.T) {
...
+	ab.SetFloat("credit_quota", 66.6)
 	q = ab.Statement()
-	a.Equal(`ALTER RESOURCE MONITOR "resource_monitor" SET CREDIT_QUOTA=66`, q)
+	a.Equal(`ALTER RESOURCE MONITOR "resource_monitor" SET CREDIT_QUOTA=66.600000`, q)
 
 	cb := snowflake.ResourceMonitor("resource_monitor").Create()
 	cb.NotifyAt(80).NotifyAt(90).SuspendAt(95).SuspendImmediatelyAt(100)
 	cb.SetString("frequency", "YEARLY")
-	cb.SetInt("credit_quota", 666)
+	cb.SetFloat("credit_quota", 666.66666666)
```
Comment (edulop91) [inline pkg/snowflake/resource_monitor_test.go]: i wonder if it is also worth keeping a test where the input "looks like an integer" and making sure the parsing works properly

File: pkg/resources/resource_monitor.go (hunk 1 above)
Comment (chauvm) [reply pkg/resources/resource_monitor.go]: done

File: pkg/snowflake/resource_monitor_test.go (hunk 2 above)
Comment (chauvm) [reply pkg/snowflake/resource_monitor_test.go]: good point

File: pkg/snowflake/resource_monitor_test.go [hunk 3]
```diff
@@ -22,14 +22,20 @@ func TestResourceMonitor(t *testing.T) {
...
 	cb.NotifyAt(80).NotifyAt(90).SuspendAt(95).SuspendImmediatelyAt(100)
 	cb.SetString("frequency", "YEARLY")
-	cb.SetInt("credit_quota", 666)
+
+	cb.SetFloat("credit_quota", 666.66666666)
+	q = cb.Statement()
+	a.Equal(`CREATE RESOURCE MONITOR "resource_monitor" FREQUENCY='YEARLY' CREDIT_QUOTA=666.666667 TRIGGERS ON 80 PERCENT DO NOTIFY ON 90 PERCENT DO NOTIFY ON 95 PERCENT DO SUSPEND ON 100 PERCENT DO SUSPEND_IMMEDIATE`, q)
+
+	// Check if credit quota can be parsed correctly to float if given an integer
+	cb.SetFloat("credit_quota", 666)
```
Comment (chauvm) [inline pkg/snowflake/resource_monitor_test.go]: Added this test, integers are converted to floats as expected

File: pkg/snowflake/generic.go [hunk 4]
```diff
@@ -81,6 +87,10 @@ func (ab *AlterPropertiesBuilder) Statement() string {
 		sb.WriteString(fmt.Sprintf(" %s=%d", strings.ToUpper(k), v))
 	}
 
+	for k, v := range ab.floatProperties {
+		sb.WriteString(fmt.Sprintf(" %s=%f", strings.ToUpper(k), v))
```
Comment (edulop91) [inline pkg/snowflake/generic.go]:
```suggestion
		sb.WriteString(fmt.Sprintf(" %s=%2f", strings.ToUpper(k), v))
```
what do you think about setting the resolution to 2 decimals?

Comment (chauvm) [reply pkg/snowflake/generic.go]: if that's what users usually provide I think it makes more sense to round up to 2 figures. I'll fix the tests and documentation to reflect that.

**Comment by ms32035** [comment]
That was quick. Thanks

**Pull request merged**